### ISSUE TITLE
small text on web site(s)

### ISSUE THREAD

**Issue by kberry** [learnlatex/learnlatex.github.io#27]
For what it's worth, the text on learnlatex (and texfaq.org) is quite small for my aging eyes. I have a large monitor and use big font sizes  to help me read, but these sites defeat these preferences and use their own idea of the best size (which I guess might be 1em, from a cursory look at the css). While leaving a quarter or so of the window width uselessly blank. The result is not unusable, but it's rather unpleasantly small for me; I've basically stopped using or contributing to texfaq because of this problem.

I don't know if it's anything you care to address, but I thought I'd at least mention it. I'll attach a screenshot, though how this image itself looks is clearly also going to depend on what it's viewed on. Anyway, as I say, FWIW ... --best, karl.

![s](https://user-images.githubusercontent.com/362855/82584185-fe065580-9b48-11ea-9f3b-a136444096b3.png)

**Comment by davidcarlisle** [comment]
@kberry  yes I guess I'm a similar age...

the white space though isn't exactly useless, it keeps the line lengths readable and in firefox and chrome at least, if you zoom in a bit (I often use 110% or 120%) it allows the line breaking to stay unchanged and the zoom level is preserved as you navigate around the site.

That said, as you know, the plan is to have the site professionally restyled once the content is done, so  we should flag this with the designer. It would be possible to have an option to zoom in more or to pick up the browser default sizes better, so we should leave this open...

**Comment by josephwright** [comment]
Hmm, coming out smaller/narrower than it does for me:
![Clipboard02](https://user-images.githubusercontent.com/654518/82603066-eb703880-9ba9-11ea-9194-019edbb89333.png)

As @davidcarlisle says, we've not done any design yet: the layout is just 'GitHub Vanilla'. Certainly one to test, etc. Could you add details of your browser, system, etc.

**Comment by kberry** [comment]
It is the same for me with all browsers I've tried - chrome, firefox,
seamonkey. CentOS 7. 4k monitor. What I suspect is most relevant is that
I have set desired font size in the various preferences to ~18
(firefox/seamonkey) or 32 (chrome, for some crazy reason it comes out
about the same as 16-18 in firefox).

It is the same with texfaq.org; I thought that had already been
designed, but I guess not.

I didn't mention it before but latex-project.org (which as I recall did
get the Jonas treatment already) is good wrt font size, so that's good;
forget about that. But it also uselessly has huge margins and thus much
too short line lengths (another screenshot attached).

I disagree that it is useful to arbitrarily increase the margins to
reduce the line length to "something reasonable". That's two different
things. Even if my entire window was used, the line length would be
completely reasonable because, gee, I defined my browser preferences so
that they work well for me. Shocking, I know :). AFAIK the "adaptive"
way to avoid ridiculously wide text lines is with max-width:64rem or
whatever number one likes, not blanking out half the window.

Forcing users to zoom on a particular site for basic readability is not
a good outcome. I'm sure you agree.

Anyway, fine. Thanks for the replies. Feel free to close this and I'll
look again when it's really ready. Sorry for jumping the gun.

**Comment by davidcarlisle** [comment]
As joseph says, the styling is almost completely unmodified github pages default theme (jekyll-theme-primer). (In fact we haven't checked any css in so you can strike the "almost" from the previous sentence:-)

Leave this open so we don't forget when drawing up the design spec. 
We could probably adjust texfaq now but it might be better to wait this site done and we can recycle some of that to texfaq to get a similar look and feel.

**Comment by davidcarlisle** [comment]
@kberry  just to confirm your report, the browser option size setting is ignored as the default css rules for `body` and `.markdown-body` (a class on a div around the whole content) both set the font size in px rather than allowing the user-set font size reflected in em or rem take effect.

As noted above this is currently just the default github pages theme and we have no CSS checked in at all. As the use of the default theme is temporary taking no action now. As it happens my default font size in firefox (16) is the same as was being forced here so I wouldn't have noticed.

As you comment, it would also probably be better to set the text body width based on the default font size so more of the screen gets used at larger sizes. (That can be tricky depending on the non textual items on the page)

If I disable the two font-size rules in the firefox inspector, (and have my default font size set to 26) it looks like

![image](https://user-images.githubusercontent.com/1268738/89915550-67e54700-dbee-11ea-85d6-c912264f9866.png)


If I also change the max-width on `.container-lg` from 1012px to   (arbitrarily chosen) 70rem it looks like

![image](https://user-images.githubusercontent.com/1268738/89915928-e0e49e80-dbee-11ea-9a87-5e9c26327713.png)

**Comment by kberry** [comment]
Thanks. Your test max-width looks much better, of course. I hope the final layout will follow something like this (and that latex-project.org gets updated). FWIW, on tug.org I use

```
BODY {
  margin-top: 1rem;
  margin-left: 1rem; /* auto results in two-digit <ol> lost off left */
  margin-right: 1rem;
  margin-bottom: 1rem;
  /* the idea is to use the whole window, unless it is ridiculously
     wide, probably with too-small fonts, too.  */
  max-width: 64rem;
}
...
/* phones, tablets, whatever: render bigger. Requires viewport decl. */
@media (max-width:768px) {
  body  { font-size:130%; }
  /* quirks mode (what we use) reverts to the browser default size
     (16px) in tables; this makes it inherit the increased body size
     that we just set: */
  table { font-size: 100%; }
}
```

(tug.org/tugstyle.css) as far as i've seen, it adapts to radically different user setups reasonably. (it took quite some time to get to that point.)

By the way, I just added learnlatex.org to tug.org/begin.html. Maybe a few people will find it there.

**Comment by kberry** [comment]
Thanks. Your test max-width looks much better, of course. I hope the final layout will follow something like this (and that latex-project.org gets updated). FWIW, on tug.org I use

```
BODY {
  margin-top: 1rem;
  margin-left: 1rem; /* auto results in two-digit <ol> lost off left */
  margin-right: 1rem;
  margin-bottom: 1rem;
  /* the idea is to use the whole window, unless it is ridiculously
     wide, probably with too-small fonts, too.  */
  max-width: 64rem;
}
...
/* phones, tablets, whatever: render bigger. Requires viewport decl. */
@media (max-width:768px) {
  body  { font-size:130%; }
  /* quirks mode (what we use) reverts to the browser default size
     (16px) in tables; this makes it inherit the increased body size
     that we just set: */
  table { font-size: 100%; }
}
```

(tug.org/tugstyle.css) as far as i've seen, it adapts to radically different user setups reasonably. (it took quite some time to get to that point.)

By the way, I just added learnlatex.org to tug.org/begin.html. Maybe a few people will find it there.

Thanks.

**Comment by kberry** [comment]
p.s. github told me my comment was not submitted. which is why it appears twice (when i expected not at all). sorry, but that's life with the web, i guess.

**Comment by jonasjacek** [comment]
@kberry All (font) sizes are currently set in rem, so they all rely on the font size of the root element. If your browser setting is default = 18, then everything will look larger. That's why I would like to know how you like the sizes of elements and texts on https://learnlatex.netlify.app/en/. I will start working on this topic soon. Thank you for your feedback.

**Comment by kberry** [comment]
Hi Jonas - the text sizes on https://learnlatex.netlify.app/en/ look
good to me! Thanks much.

I have two comments, though (maybe you haven't worked on these items
yet, but since I'm looking at it):

1) the content uses only approximately 1/3 of my (desktop) browser
window; in other words, the "margins" are absurdly big, and the line
length absurdly short - maybe 40 chars, which seems ok for mobile, but
way tiny for desktop.

2) Item 9 breaks across columns, I.e., starts at the bottom of column 1
and ends at the top of column 2.

Both of these things are happening for me in both seamonkey and firefox,
on CentOS 7.

I will attach a screenshot. (I wonder if github will accept it.)

Thanks,
Karl

**Comment by davidcarlisle** [comment]
Karl, with my normal setting the item doesn't break and the character count per line is reasonable, but I see both the things you mention if I up the default browser font size, I think it's just this one rule that still has px units and wants to be  in rem (but not more than body width:-) I'm sure Jonas will get to it, but just confirming the issue, I'll add a screenshot, as yours didn't seem to make it

```
.wrapper {
  max-width: 600px;
  margin: 0 auto;
  padding: 0 0.5rem;
}
```

![image](https://user-images.githubusercontent.com/1268738/98273103-c91c7c80-1f89-11eb-8402-b73ec2ac18fd.png)

and the same with max-width changed to 50rem

![image](https://user-images.githubusercontent.com/1268738/98273443-2dd7d700-1f8a-11eb-981c-eb07c0f4e2b0.png)

**Comment by jonasjacek** [comment]
Thank you both of you. I am working on the wrapper and the contents, yes. 
I am still unsure how to manage characters and words per line, if I make the wrapper wider. Breaking content up into columns may work well on the home page but not e.g. on lesson pages.

I also kept in mind, that many users split their screen, so a browser window might take just half of the existing space. So, this topic is still open.

The lessons table of content on the home page will most likely be refactored for using HTML flexbox. The current implementation (just a two column list) is too unstable and might not work well for other languages anyway. I will do this, once I am happy with the content part. The lesson table is completely new and not translated at all. My focus was on helping users to scan the list (users barely read once they find what they are looking for) while adding more information which can be used by e.g. Search Engines to better weigh and rank the linked resources.

Again, thank you all for your feedback.

**Comment by jonasjacek** [comment]
Closing this ticket, as the font-size issue seems to be fixed. If the other issues persist, please open a new ticket at a later point. Thanks.

**Comment by davidcarlisle** [comment]
This issue has been closed but the font setting appears to have reverted. In teh design branch

```
/*----------------------------------------------------------------------------*/
/*  # ROOT ELEMENT: DEFAULT FONT SIZE                                         */
/*----------------------------------------------------------------------------*/
:root {
  font-size: 18px;
}
```

is forcing 18px and over-riding the users browser default size.

**Comment by jonasjacek** [comment]
I've closed this ticket, because the root font-size was increased by 2px (from 16 to 18). This is 2px larger than the default paragraph font-size on latex-project.org and 1px larger than on e.g. theguardian.com. With that change, I've also made the wrapper 60px wider. 

The chars per line should be ([UX best practice](https://www.google.com/search?q=ux+optimal+characters+per+line)) 50-75 chars. We have more than 85 per line.

---

However, I've just [pushed](https://github.com/learnlatex/learnlatex.github.io/commit/71e5bbb6ddb319549f3ba81de5c2a4609c460cbd) the styles, suggested by @kberry, without adding . Can be seen now at: https://learnlatex.netlify.app/en/

You decide. It's all responsive. Changing things should not take too many adjustment.

**Comment by davidcarlisle** [comment]
@jonasjacek  As we have no graphical elements to match, I think Karl is right that we should honour the users default font size, but I think we should still constrain the max-width of the body to ~60 em or whatever looks right. Note that the body width before the last change was probably too tight (or the editor font too large) as  several examples (especially in lesson 10) had acquired horizontal scrollbars which we should avoid (all code examples have less than 80 columns)

**Comment by jonasjacek** [comment]
You are absolutely right about letting the user decide about font-sizes. I've changed that. 
I've also changed the default width to 50rem. To my eyes, this is the limit before text runs way too wide to be read comfortably. 50 rem covers ~97 columns of text within the ace editor.
To cover 80 columns without an horizontal scrollbar, 42rem (based on 16px default font size) should do as default width.

**Comment by josephwright** [comment]
How does it look now?

**Comment by kberry** [comment]
looks nice!!

**Comment by car222222** [comment]
Maybe I could add a bit more to think about on readability:
in long paragraphs the use of a very thin (light weight?) font, combined with its being sans serif, causes difficulties for my visual system.  

Somehow, the whole page seems 'lost in a mist' with very low contrast. Can we not afford more black pixels for maximal impact and legibility? :-)

**Comment by jonasjacek** [comment]
> Maybe I could add a bit more to think about on readability:
> in long paragraphs the use of a very thin (light weight?) font, combined with its being sans serif, causes difficulties for my visual system.
> 
> Somehow, the whole page seems 'lost in a mist' with very low contrast. Can we not afford more black pixels for maximal impact and legibility? :-)

I will try to brush up a little more, once there is more feedback. 

* Contrasts: we are already WCAG AAA compliant. 
* Paragraph length: has been discussed in this thread before. https://github.com/learnlatex/learnlatex.github.io/issues/27#issuecomment-751706256

**Comment by car222222** [comment]
Good to know: Contrasts: we are already WCAG AAA compliant.

How is this defined and determined?  
Does it take into account different screen capabilities and different eyesight capabilities?  

As you say, further feedback is needed, and is more use than automatic verification.

**Comment by car222222** [comment]
I did not find any discussion of paragraph length, only if line length.

But this is irrelevant since I was not talking about changing paragraph, or line, length.  

I was just giving an example of one case when the lack of contrast and use of sans serif makes it all go rather fuzzy to me.

**Comment by josephwright** [comment]
I think we are OK here now
